verify.go

```diff
@@ -17,53 +17,53 @@ import (
 // IDTokenVerifier provides verification for ID Tokens.
 type IDTokenVerifier struct {
 	keySet *remoteKeySet
-	config *verificationConfig
-}
-
-// verificationConfig is the unexported configuration for an IDTokenVerifier.
-//
-// Users interact with this struct using a VerificationOption.
-type verificationConfig struct {
+	config *Config
 	issuer string
-	// If provided, this value must be in the ID Token audiences.
-	audience string
-	// If not nil, check the expiry of the id token.
-	checkExpiry func() time.Time
-	// If specified, only these sets of algorithms may be used to sign the JWT.
-	requiredAlgs []string
-	// If not nil, don't verify nonce.
-	nonceSource NonceSource
 }
 
-// VerificationOption provides additional checks on ID Tokens.
-type VerificationOption interface {
-	// Unexport this method so other packages can't implement this interface.
-	updateConfig(c *verificationConfig)
+// Config is the configuration for an IDTokenVerifier.
+type Config struct {
+	// The ID of the client that started the login flow, which is expected to be in the ID Token's audiences.
```

Comment:
> The ID of the client that started the login flow,

This isn't always true. We probably want something like

> Expected audience of the token. For a majority of the cases this is expected to be the ID of
> the client that initialized the login flow. It may occasionally differ if the provider supports the
> authorizing party (azp) claim.
>
> If not provided, users must explicitly set SkipClientIDCheck.